### PR TITLE
WINDUP-569: Add a prescan filter for TypeInterestFactory and remove unec...

### PR DESCRIPTION
**Pull request by jsight**
...essary work from indexing class files

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/62/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/691/<h2>Failed Tests: <span class='status-failure'>2</span></h2><h3><a name='org.jboss.windup.rules.apps:windup-rules-java-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/691/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport'>org.jboss.windup.rules.apps:windup-rules-java-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/691/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassCompositeTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassCompositeTest.testJavaClassCondition</strong></a></li></ul><h3><a name='org.jboss.windup.tests:windup-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/691/org.jboss.windup.tests$windup-tests/testReport'>org.jboss.windup.tests:windup-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/691/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSmallBinaryMode2Test/testRunWindupTiny/'><strong>org.jboss.windup.tests.application.WindupArchitectureSmallBinaryMode2Test.testRunWindupTiny</strong></a></li></ul>
Test FAILed.

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/AddClassFileMetadata.java [hunk 1]
```diff
@@ -48,83 +46,81 @@ public void perform(GraphRewrite event, EvaluationContext context, JavaClassFile
...
+                String packageAsPath = excludePackage.getPackageName().replace(".", "/");
+                if (absoluteFile.contains(packageAsPath))
+                    return;
+            }
+
+            boolean shouldScan = true;
+            for (PackageModel includePackage : javaConfiguration.getScanJavaPackages())
+            {
+                String packageAsPath = includePackage.getPackageName().replace(".", "/");
+                if (absoluteFile.contains(packageAsPath))
```
Comment (bradsdavis) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/AddClassFileMetadata.java]: I would think you should be using startsWith here.

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/AddClassFileMetadata.java [hunk 2]
```diff
@@ -48,83 +46,81 @@ public void perform(GraphRewrite event, EvaluationContext context, JavaClassFile
...
-                        for (final String interfaceName : interfaceNames)
-                        {
-                            JavaClassModel interfaceModel = javaClassService.getOrCreatePhantom(interfaceName);
-                            javaClassModel.addImplements(interfaceModel);
-                        }
-                    }
+                String simpleName = qualifiedName;
+                if (packageName != null && !packageName.equals("") && simpleName != null)
+                {
+                    simpleName = simpleName.substring(packageName.length() + 1);
```
Comment (bradsdavis) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/AddClassFileMetadata.java]: We have Commons Lang.  Any reason not to use StringUtils.removeStart?  It reads more clearly.

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/AddClassFileMetadata.java (hunk 1 above)
Comment (jsight) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/AddClassFileMetadata.java]: That won't work if the base for the package is a subdirectory, and I don't want to have special logic for all of the cases where it is not.

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/AddClassFileMetadata.java (hunk 2 above)
Comment (jsight) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/AddClassFileMetadata.java]: +1

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/64/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/693/
Test PASSed.

**Pull request merged**